### PR TITLE
Add message codes for BuildCheck suggestion diags

### LINKED ISSUE #10501: BuildCheck suggestions do not start with code
### Context

Compare resuts for `BC0202` and `BC0203`

```
Set Property: MyPropT2=;xxx
MSBUILD : warning BC0202: C:\Users\jankrivanek\AppData\Local\Temp\wfac2uyx.xnb\MSBuildTemp\Temporaryfcdcb69ef02b4c23ab44afca5dc462b1\myProj.proj (24,10): https://aka.ms/buildcheck/codes#BC0202 - Property: [MyPropT2] first declared/initialized at [C:\Users\jankrivanek\AppData\Local\Temp\wfac2uyx.xnb\MSBuildTemp\Temporaryfcdcb69ef02b4c23ab44afca5dc462b1\myProj.proj (26,1)] used before it was initialized.
Done building target "PrintEnvVar" in project "myProj.proj".
C:\Users\jankrivanek\AppData\Local\Temp\wfac2uyx.xnb\MSBuildTemp\Temporaryfcdcb69ef02b4c23ab44afca5dc462b1\myProj.proj (13,3): https://aka.ms/buildcheck/codes#BC0203 - Property: [MyProp13] was declared/initialized, but it was never used.
MSBUILD : warning BC0201: C:\Users\jankrivanek\AppData\Local\Temp\wfac2uyx.xnb\MSBuildTemp\Temporaryfcdcb69ef02b4c23ab44afca5dc462b1\myProj.proj (13,3): https://aka.ms/buildcheck/codes#BC0201 - Property: [MyProp11] was accessed, but it was never initialized.
```

Since `BC0203` has the severity set to `Suggestion` it doesn't contain the code at the start of the output

### Expected behavior

Suggestion, Warning and Error has similar output fomratting

### PR DESCRIPTION
**Pull request by JanKrivanek**
Fixes #10501

### Context

BuildCheck suggestions didn't have the codes and formatting of warnings and errors.

### Changes Made

Unified the diagnostics creation code

### CODE REVIEW

**Review by maridematte — COMMENTED** [review]
Will this changes need to be reflected on the binlog viewer too? I know the process of adding to it but not when we're removing something.

**Comment by JanKrivanek** [comment]
> Will this changes need to be reflected on the binlog viewer too? I know the process of adding to it but not when we're removing something.

Good question.

tl;dr;: it's fine here :-)

In this specific case we are not removing any data definition - the binlog OM stays same.
We are removing couple methods writing a specific types of data to the binlog - 2 of them were already dead (writing BuildCheck error/warning) and the third was used - but there was no reason for differentiating BuildCheck message in a binlog, while BuildCheck error nor warnings were differentiated.

**Review by YuliiaKovalova — APPROVED** [review]
looks good to me

File: src/Build/BuildCheck/API/BuildCheckResult.cs [hunk 1]
```diff
@@ -29,13 +29,15 @@ internal BuildEventArgs ToEventArgs(CheckResultSeverity severity)
         => severity switch
         {
             CheckResultSeverity.Suggestion => new BuildCheckResultMessage(this),
-            CheckResultSeverity.Warning => new BuildCheckResultWarning(this, CheckRule.Id),
-            CheckResultSeverity.Error => new BuildCheckResultError(this, CheckRule.Id),
+            CheckResultSeverity.Warning => new BuildCheckResultWarning(this),
+            CheckResultSeverity.Error => new BuildCheckResultError(this),
             _ => throw new ArgumentOutOfRangeException(nameof(severity), severity, null),
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Build/BuildCheck/API/BuildCheckResult.cs:34]: `ArgumentOutOfRangeException` is the choice here
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression

File: src/Framework/BuildMessageEventArgs.cs [hunk 2]
```diff
@@ -109,6 +109,28 @@ public BuildMessageEventArgs(
...
+        /// <param name="message">text message</param>
+        protected BuildMessageEventArgs(
+            string code,
+            string message,
+            string file,
+            int lineNumber,
+            int columnNumber)
+            : base(message, helpKeyword: null, senderName: null)
+        {
+            this.code = code;
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Framework/BuildMessageEventArgs.cs:128]: It's a legit situation. Then `code` member field will be null - same as if different constructor, that doesn't initialize it, would be called
